mesonbuild/interpreter.py

```diff
@@ -3761,7 +3761,7 @@ def unpack_env_kwarg(self, kwargs) -> build.EnvironmentVariables:
 
     def add_test(self, node, args, kwargs, is_base_test):
         if len(args) != 2:
-            raise InterpreterException('Incorrect number of arguments')
+            raise InterpreterException(r'test expects 2 arguments, only {len(args)} given')
```

Comment:
This isn't doing what you're intending it to do. the `r` prefix in python makes a string a `raw string`. Python assumes you mean exactly what you see (no escaping, no unicode interpretation, etc). What I think you meant to do is use an `f` prefix, an `f-string` does inline interpolation, ie:

```python
>>> a = 'foo'
>>> print(f'I have {a}')
I have foo
>>> print(r'I have {a}')
I have {a}
```

Unfortunately f-strings are a python 3.6 feature and meson still needs to work with 3.5, so you'll need to use `str.format()` instead:

```python
raise InterpreterException(r'test expects 2 arguments, {} given'.format(len(args))
```